src/UniversalRewardsDistributor.sol

```diff
@@ -98,9 +97,9 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
         root = pendingRoot.root;
         ipfsHash = pendingRoot.ipfsHash;
 
-        emit EventsLib.RootSet(pendingRoot.root, pendingRoot.ipfsHash);
+        _setPendingRoot(PendingRoot(0, 0, 0));
 
-        delete pendingRoot;
+        emit EventsLib.RootSet(root, ipfsHash);
```

Comment:
Can we have this event emitted right after the `root` and `ipfsHash` assignment? Before `_setPendingRoot`